File: pgm/ABC_Kotlin/ABC_365_A.kt
// import java.time.*

import kotlin.math.*

const val four: Int = 4
const val hundred: Int = 100
const val year: Int = 365
const val leapYear: Int = 366
const val fourHundred: Int = 400
const val zero: Int = 0

fun main() {
    val Y: Int = readln().toInt()
    val isLeapYear: Boolean =
            when {
                Y % four != zero -> false
                Y % four == zero && Y % hundred != 0 -> true
                Y % hundred == zero && Y % fourHundred != 0 -> false
                else -> true
            }
    if (isLeapYear) {
        println(leapYear)
    } else {
        println(year)
    }
    kotlin.system.exitProcess(0)
}

File: pgm/ABC_Dart/ABC_365_A.dart
// import 'dart:collection';
import 'dart:io';
// import 'dart:math';
// import 'dart:typed_data';

void main() {
  final int Y = int.parse(stdin.readLineSync()!);
  const int four = 4;
  const int hundred = 100;
  const int year = 365;
  const int leapYear = 366;
  const int fourHundred = 400;
  const int zero = 0;
  if (Y % four != zero) {
    print(year);
  } else if (Y % four == zero && Y % hundred != zero) {
    print(leapYear);
  } else if (Y % hundred == zero && Y % fourHundred != zero) {
    print(year);
  } else {
    print(leapYear);
  }
  exit(0);
}
